.gitlab-ci.yml

```diff
@@ -44,11 +44,22 @@ build:
 ###############################################################
 
 test:
+  stage: test
   tags:
     - kube
+  before_script:
+    - docker pull $REPO_URL/$BASE_IMAGE_NAME:$CI_COMMIT_SHA
+  script:
+    - docker run $REPO_URL/$BASE_IMAGE_NAME:$CI_COMMIT_SHA pip install -r requirements.dev.txt; make test
+
+test-mypy:
   stage: test
+  tags:
+    - kube
+  before_script:
+    - docker pull $REPO_URL/$BASE_IMAGE_NAME:$CI_COMMIT_SHA
   script:
-    - echo "Still to be implemented"
+    - docker run $REPO_URL/$BASE_IMAGE_NAME:$CI_COMMIT_SHA pip install -r requirements.dev.txt; make mypy
```

Comment:
`make mypy` is the only thing that I'm on the fence about, there's some rough edges and I want us to have a well-defined escape hatch if we need to push code we believe to be correct but that `mypy` can't stop complaining 
about.

I think at the minimum we could just add something about  `# mypy: ignore-errors`

Reply:
I changed it to allow failures on mypy